modules/dxp/apps/commerce/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/diagram/components/Form.js

```diff
@@ -14,6 +14,20 @@ import Autocomplete from 'commerce-frontend-js/components/autocomplete/Autocompl
 import React from 'react';
 
 export function LinkedToCatalogProductFormGroup({updateValue, value}) {
+
+	/**
+	 * Product model doesn't match with MappedProduct.
+	 * When the product details come from pins APIs,
+	 * we need to align both the models in order
+	 * to provide consistent data to the Autocomplete
+	 */
+
+	const initialValue = value ? {...value} : null;
```

Comment:
What's the intent on this line? :thinking: If `value` is object, this only sets falsy values to `null`. If `value` is not an object, and is for example string or number, this will have probably unwanted spreading effect. 

The same for L69.

Reply:
If you want a variable that doesn't re-initialize with component update, you should use a ref:
```js
const valueRef = useRef(value);
```